stan/math/prim/mat/err/check_finite.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+
+#include <Eigen/Dense>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+
+namespace stan {
+namespace math {
+namespace {
```

Comment:
We decided to use the `internal` namespace rather than anonymous namespaces because the anonymous namespaces didn't seem to do what they advertised (file private scope).  Here, it looks like you're specializing another structure in another anonymous namespace!